tests/conftest.py

```diff
@@ -1,42 +1,139 @@
 # SPDX-License-Identifier: MIT
 
+import os
+import os.path
+import shutil
+import stat
+import sys
+import tarfile
+import tempfile
+
+import filelock
 import pytest
 
+import build
+
+
+if sys.version_info[0] == 2:
+    from urllib2 import urlopen
+else:
+    from urllib.request import urlopen
+
 
-DUMMY_PYPROJECT = '''
-[build-system]
-requires = ['flit_core']
-build-backend = 'flit_core.buildapi'
-'''.strip()
+INTEGRATION_SOURCES = {
+    'dateutil': ('dateutil/dateutil', '2.8.1'),
+    'pip': ('pypa/pip', '20.2.1'),
+    'Solaar': ('pwr-Solaar/Solaar', '1.0.3'),
+}
 
 
-DUMMY_PYPROJECT_NO_BACKEND = '''
-[build-system]
-requires = []
-'''.strip()
+def pytest_addoption(parser):
+    parser.addoption('--run-integration', action='store_true', help='run the integration tests')
+    parser.addoption('--only-integration', action='store_true', help='only run the integration tests')
 
 
-DUMMY_PYPROJECT_NO_REQUIRES = '''
-[build-system]
-build-backend = 'something'
-'''.strip()
+def pytest_collection_modifyitems(config, items):
+    skip_int = pytest.mark.skip(reason='integration tests not run')
+    skip_other = pytest.mark.skip(reason='only integration tests are run')
+
+    for item in items:
+        if os.path.basename(item.location[0]) == 'test_integration.py':
+            if not config.getoption('--run-integration') and not config.getoption('--only-integration'):
+                item.add_marker(skip_int)
+        elif config.getoption('--only-integration'):
+            item.add_marker(skip_other)
 
 
 @pytest.fixture
-def empty_file_mock(mocker):
-    return mocker.mock_open(read_data='')
+def packages_path():
+    return os.path.realpath(os.path.join(__file__, '..', 'packages'))
+
+
+@pytest.fixture(scope='session')
+def integration_path():
+    srcdir = os.path.realpath('.integration-sources')
+    dest = tempfile.mkdtemp(prefix='python-build-integration-')
+
+    if not os.path.exists(srcdir):
+        os.makedirs(srcdir)
+
+    # self
```

Comment:
I'm failing to interpret this comment.

Reply:
We are handling the python-build source, hence `self`. We treat python-build differently since we use our own source instead of downloading the tarball.

Reply:
This is totally not clear from that one line, maybe put a more long winded explanation of above? 👍

Reply:
Yes, I have changed the comment, I will push soon, I am just finishing handling the other comments.